src/Adapter/SpecificPrice/CommandHandler/AddSpecificPriceHandler.php

```diff
@@ -0,0 +1,122 @@
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\SpecificPrice\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\SpecificPrice\AbstractSpecificPriceHandler;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\Command\AddSpecificPriceCommand;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\CommandHandler\AddSpecificPriceHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\Exception\SpecificPriceConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\Exception\SpecificPriceException;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\ValueObject\SpecificPriceId;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime;
+use PrestaShopException;
+use SpecificPrice;
+
+/**
+ * Handles AddSpecificPriceCommand using legacy object model
+ */
+final class AddSpecificPriceHandler extends AbstractSpecificPriceHandler implements AddSpecificPriceHandlerInterface
+{
+    /**
+     * @param AddSpecificPriceCommand $command
+     *
+     * @return SpecificPriceId
+     *
+     * @throws SpecificPriceConstraintException
+     * @throws SpecificPriceException
+     */
+    public function handle(AddSpecificPriceCommand $command): SpecificPriceId
+    {
+        try {
+            $specificPrice = $this->createSpecificPriceFromCommand($command);
+
+            if (false === $specificPrice->validateFields(false)) {
+                throw new SpecificPriceConstraintException('Specific price contains invalid field values');
+            }
+
+            if (!$specificPrice->add()) {
+                throw new SpecificPriceException(
+                    'Failed to add new specific price'
+                );
+            }
+        } catch (PrestaShopException $e) {
+            throw new SpecificPriceException('An error occurred when trying to add new specific price');
+        }
+
+        return new SpecificPriceId($specificPrice->id);
+    }
+
+    /**
+     * Creates legacy SpecificPrice object from command
+     *
+     * @param AddSpecificPriceCommand $command
+     *
+     * @return SpecificPrice
+     *
+     * @throws PrestaShopException
+     * @throws SpecificPriceConstraintException
+     */
+    private function createSpecificPriceFromCommand(AddSpecificPriceCommand $command): SpecificPrice
+    {
+        $specificPrice = new SpecificPrice();
+
+        $specificPrice->id_product = $command->getProductId()->getValue();
+        $specificPrice->reduction_type = $command->getReduction()->getType();
+        $specificPrice->reduction = $command->getReduction()->getValue();
+        $specificPrice->reduction_tax = $command->isIncludeTax();
```

Comment:
No cast ? Both are booleans ?

Reply:
yes both are booleans. The isIncludeTax() is typecasted with bool. And the reduction_tax is tinyInt type in db, so object model handles bool well too.